src/harness/testcases/WINNF_FT_S_FAD_testcase.py

```diff
@@ -0,0 +1,110 @@
+        self._sas_admin.InjectFccId({'fccId': device_c['fccId']})
+        request = {'registrationRequest': [device_a, device_c]}
+        response = self._sas.Registration(request)['registrationResponse']
+        # Check registration response
+        cbsd_ids = []
+        for resp in response:
+            self.assertEqual(resp['response']['responseCode'], 0)
+            cbsd_ids.append(resp['cbsdId'])
+        del request, response
+
+        # Create grant requests
+        grant_0 = json.load(
+            open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+        grant_1 = json.load(
+            open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+        grant_0['cbsdId'] = cbsd_ids[0]
+        grant_1['cbsdId'] = cbsd_ids[1]
+        # Request for non-overlapping frequency spectrum
+        grant_0['operationParam']['operationFrequencyRange'] = {
+            'lowFrequency': 3620000000.0,
+            'highFrequency': 3630000000.0
+        }
+        grant_1['operationParam']['operationFrequencyRange'] = {
+            'lowFrequency': 3640000000.0,
+            'highFrequency': 3650000000.0
+        }
+        request = {'grantRequest': [grant_0, grant_1]}
+        # Send grant requests
+        response = self._sas.Grant(request)['grantResponse']
+        # Check registration response
+        grant_ids = []
+        for resp in response:
+            self.assertEqual(resp['response']['responseCode'], 0)
+            grant_ids.append(resp['grantId'])
+        del request, response
+
+        # Trigger the SAS to generate the activity dump
+        self._sas_admin.TriggerFullActivityDump()
+
+        # Get dump 
+        response = self._sas.GetFullActivityDump()
+        # Verify the response with  FullActivityDump.schema.json Object schema
+        self.assertContainsRequiredFields("FullActivityDump.schema.json", response)
+        generation_date_time = datetime.strptime(response['generationDateTime'],
+                                          '%Y-%m-%dT%H:%M:%SZ')
+        if generation_date_time >= (datetime.utcnow() - timedelta(days=7)):
+            for file in response['files']:
+                # Verify the response files with ActivityDumpFile.schema.json Object schema
+                self.assertContainsRequiredFields("ActivityDumpFile.schema.json", file)
+                # Verify the url format is valid (pip install validators)
+                self.assertTrue(validators.url(file['url']))
+                # Verify the link to download the file dump is valid
+                self.assertEqual(urlopen(file['url']).getcode(), 200)
+                # Verify the record type 
+                self.assertEqual(file['recordType'], 'cbsd')
+                #get data in file
+                data = urlopen(file['url']).read()
+                #Decoding JSON message from file 
+                data = json.loads(data)
+                for record in data['recordData']:
```

Comment:
The checks below verify that everything in the full dump matches a CBSD or Grant created at the beginning of the test script, but it does not check that *all* CBSDs and Grants created are present in the full dump. For example, a full dump response which is empty would pass this test.